app/apollo/resolvers/cluster.js

```diff
@@ -577,11 +677,11 @@ const clusterResolvers = {
       const user = whoIs(me);
 
       try {
-        logger.info({ req_id, user, org_id, registration }, `${queryName} validating`);
+        logger.info({req_id, user, org_id, registration}, `${queryName} validating`);
 
-        await validAuth(me, org_id, ACTIONS.REGISTER, TYPES.CLUSTER, queryName, context);
+        await validAuth(me, org_id, ACTIONS.REGISTER, TYPES.CLUSTER, queryName, context, [UUID(), registration.name]);
```

Comment:
Why passing `UUID()` as a resource value to check auth?

Reply:
There isn't a defined uuid at this point. It is created right after checking auth. We want to check if user is allowed to create this before creating the resource.

Reply:
Just making up a new UUID and checking if the user has allowed to create that random UUID doesn't add value though.
```suggestion
        await validAuth(me, org_id, ACTIONS.REGISTER, TYPES.CLUSTER, queryName, context, [registration.name]);
```

Reply:
Commited this change for each resolver